MediaBrowser.Api/Playback/Progressive/BaseProgressiveStreamingService.cs

```diff
@@ -356,10 +363,31 @@ private async Task<object> GetStreamResult(StreamRequest request, StreamState st
             var contentType = state.GetMimeType(outputPath);
 
             // TODO: The isHeadRequest is only here because ServiceStack will add Content-Length=0 to the response
+            var contentLength = state.EstimateContentLength || isHeadRequest ? GetEstimatedContentLength(state) : null;
+
+            if (contentLength.HasValue)
+            {
+                responseHeaders[HeaderNames.ContentLength] = contentLength.Value.ToString(CultureInfo.InvariantCulture);
+            }
+
             // Headers only
             if (isHeadRequest)
             {
-                return ResultFactory.GetResult(null, Array.Empty<byte>(), contentType, responseHeaders);
+                var streamResult = ResultFactory.GetResult(null, Array.Empty<byte>(), contentType, responseHeaders);
+
+                if (streamResult is IHasHeaders hasHeaders)
+                {
+                    if (contentLength.HasValue)
+                    {
+                        hasHeaders.Headers[HeaderNames.ContentLength] = contentLength.Value.ToString(CultureInfo.InvariantCulture);
+                    }
+                    else if (hasHeaders.Headers.ContainsKey(HeaderNames.ContentLength))
```

Comment:
You don't need to check if it exists before removing it. (Unless it's some custom improperly implemented dictionary.